schema/object.go

```diff
@@ -684,6 +652,41 @@ func (a *AnyTypedObject[T]) Any() TypedObject[any] {
 	return a
 }
 
+// ConvertToObjectSchema attempts to extract the given type or data from a
+// RefSchema or from the root of a ScopeSchema, and then cast it to an
+// ObjectSchema. If successful it returns true, else it returns false.
```

Comment:
I think this could use more polish.

I'd do something like:
> ConvertToObjectSchema attempts to extract an ObjectSchema from the input.
> - If an ObjectSchema is found, it returns it.
> - If a RefSchema is found, it extracts the cached object schema the ref is referencing.
> - If a ScopeSchema is found, it extracts the root object schema.
> Returns true if successful, false otherwise.

Reply:
ah, I generally try not to write something that close to a pseudocode algorithm in the function documentation string, but I'll change it.

Reply:
Yeah. It does explain a lot of the internals. But for this function, I feel like we need to specify the specific cases.